Engine/Client/Src/Application/Application.cpp

```diff
@@ -45,83 +48,149 @@ void Application::run()
-
-        for (int index = 0; index < buttonNetworkConnection.size(); index++) {
-            std::shared_ptr<NetworkConnectionComponent> networkInfo = std::dynamic_pointer_cast<NetworkConnectionComponent>(buttonNetworkConnection[index]);
-            if (!networkInfo)
-                continue;
-            if (networkInfo->connect == true) {
-                networkInfo->connect = false;
-                _sceneManager->changeScene(std::make_pair<size_t, std::string>(0, "firstScene.json"));
-                 _client = std::make_shared<Network::Client>(networkInfo->serverIp, std::atoi(networkInfo->serverPort.c_str()), 4445);
-                _client->connect([this](Network::UDPPacket packet, ECS::Registry& reg) {
-                    this->_packetHandler(std::move(packet), *_registry);
-                }, *_registry);
-            }
+    if (_client->getToken() == 0)
+        return;
+    std::lock_guard<std::mutex> lock(_client->getMutex());
+    std::vector<ABINetwork::UDPPacket> messages = _client->getReceivedMessages();
+
+    for (auto packet : messages) {
+        try {
+            auto messageType = static_cast<ABINetwork::IMessage::MessageType>(packet.getMessageType());
+            _handlePacketsMap[ABINetwork::IMessage::MessageType(messageType)](packet);
+        } catch (const std::exception &e) {
+            std::cerr << e.what() << std::endl;
         }
-        if (_client == nullptr)
-            return;
-        if (_client->getIdxPlayerComponent() != -1 && _client->getIdxPlayerComponent() < players.size()) {
-            std::shared_ptr<PlayerComponent> player = std::dynamic_pointer_cast<PlayerComponent>(players[_client->getIdxPlayerComponent()]);
+    }
+}
 
-            if (player && !player->isAlive) {
-                player->isAlive = true;
-                _sceneManager->changeScene(std::make_pair<size_t, std::string>(0, "endScene.json"));
-            }
-        }
-    } catch(const std::exception& e) {
+void Application::_handleCreateRoomPacket(ABINetwork::UDPPacket packet)
+{
+    std::tuple<std::string, int, int> roomCreated = ABINetwork::getCreatedRoomInfoFromPacket(packet);
+    _roomInfos.tcpPort = std::get<1>(roomCreated);
+    _roomInfos.udpPort = std::get<2>(roomCreated);
+    ABINetwork::sendPacketJoinRoom(_client, std::get<0>(roomCreated), ABINetwork::getCurrentRoomPassword(_client));
+}
+
+void Application::_handleJoinRoomPacket(ABINetwork::UDPPacket packet)
+{
+    if (_roomInfos.tcpPort <= 0 || _roomInfos.udpPort <= 0)
+        return;
+    std::shared_ptr<ABINetwork::INetworkUnit> room = nullptr;
+    try {
+        room = ABINetwork::createClient();
+        if (!room || !ABINetwork::connectToServer(room, "127.0.0.1", _roomInfos.tcpPort))
+            throw ClientError("Error while joining room");
+        _sceneManager->changeScene(std::make_pair<std::size_t, std::string>(0, FIRST_GAME_SCENE));

```

Comment:
For me, any content of the game can't be defined in the engine part, so here, you can change to the main game scene in the main app of the client even if it's easier

Reply:
This is the main file of the game.
We already discussed about that. It's like a main file in cpp. It must be the "firstScene.json".

Reply:
Fine, maybe I badly explained myself